meteor/server/api/rest/api.ts

```diff
@@ -0,0 +1,660 @@
+import Koa from 'koa'
+import cors from '@koa/cors'
+import KoaRouter from '@koa/router'
+import { logger } from '../../logging'
+import { WebApp } from 'meteor/webapp'
+import { check, Match } from '../../../lib/check'
+import { Meteor } from 'meteor/meteor'
+import { ClientAPI } from '../../../lib/api/client'
+import { getCurrentTime, getRandomString, protectString } from '../../../lib/lib'
+import { RestAPI, RestAPIMethods } from '../../../lib/api/rest'
+import { registerClassToMeteorMethods, ReplaceOptionalWithNullInMethodArguments } from '../../methods'
+import { RundownPlaylists } from '../../../lib/collections/RundownPlaylists'
+import { MeteorCall, MethodContextAPI } from '../../../lib/api/methods'
+import { ServerClientAPI } from '../client'
+import { ServerRundownAPI } from '../rundown'
+import { triggerWriteAccess } from '../../security/lib/securityVerify'
+import { ExecuteActionResult, StudioJobs } from '@sofie-automation/corelib/dist/worker/studio'
+import { CURRENT_SYSTEM_VERSION } from '../../migration/currentSystemVersion'
+import {
+	AdLibActionId,
+	BucketAdLibId,
+	PartId,
+	PieceId,
+	RundownBaselineAdLibActionId,
+	RundownPlaylistId,
+	SegmentId,
+	StudioId,
+} from '@sofie-automation/corelib/dist/dataModel/Ids'
+import { AdLibPieces } from '../../../lib/collections/AdLibPieces'
+import { AdLibActions } from '../../../lib/collections/AdLibActions'
+import { RundownBaselineAdLibPieces } from '../../../lib/collections/RundownBaselineAdLibPieces'
+import { RundownBaselineAdLibActions } from '../../../lib/collections/RundownBaselineAdLibActions'
+import { BucketAdLibs } from '../../../lib/collections/BucketAdlibs'
+import { UserError, UserErrorMessage } from '@sofie-automation/corelib/dist/error'
+import { StudioContentWriteAccess } from '../../security/studio'
+import { ServerPlayoutAPI } from '../playout/playout'
+import { TriggerReloadDataResponse } from '../../../lib/api/userActions'
+
+const REST_API_USER_EVENT = 'rest_api'
+
+class ServerRestAPI extends MethodContextAPI implements ReplaceOptionalWithNullInMethodArguments<RestAPI> {
+	async index(): Promise<ClientAPI.ClientResponse<{ version: string }>> {
+		triggerWriteAccess()
+
+		return ClientAPI.responseSuccess({ version: CURRENT_SYSTEM_VERSION })
+	}
+	async activate(
+		rundownPlaylistId: RundownPlaylistId,
+		rehearsal: boolean,
+		connection: Meteor.Connection
+	): Promise<ClientAPI.ClientResponse<void>> {
+		return ServerClientAPI.runUserActionInLogForPlaylistOnWorker(
+			{ ...this, connection: connection },
+			REST_API_USER_EVENT,
```

Comment:
I think this should contain some information about the request that caused this to execute, so that this provides the same debugging value as the current "user event summary". Request URL maybe?

Reply:
I've made a method that constructs a string with some useful info. Open to suggestions for better formats for this string. https://github.com/evs-broadcast/sofie-core/blob/66d52dc36c0b2e4b0bcf768261fd35fe9218aa9b/meteor/server/api/rest/api.ts#L42